src/Components/DropDownMenu.tsx

```diff
@@ -0,0 +1,94 @@
+import { Person } from '../types/Person';
+
+import React, { FC, useCallback, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+interface Props {
+  people: Person[];
+  delay?: number;
+  changeChosenPerson: (person: Person | null) => void;
+  changeAppliedQuery: (value: string) => void;
+}
+
+export const DropDownMenu: FC<Props> = ({
+  people,
+  delay = 300,
+  changeChosenPerson,
+  changeAppliedQuery,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(debounce(changeAppliedQuery, delay), []);
```

Comment:
you can include changeAppliedQuery and delay in the dependency array, this ensures that the debounce function updates correctly if changeAppliedQuery or delay changes.
```suggestion
  const applyQuery = useCallback(debounce(changeAppliedQuery, delay), [changeAppliedQuery, delay]);
```